python/cuml/test/test_linear_model.py

```diff
@@ -488,3 +488,82 @@ def test_logistic_predict_convert_dtype(train_dtype, test_dtype):
     clf = cuLog()
     clf.fit(X_train, y_train)
     clf.predict(X_test.astype(test_dtype))
+
+
+@pytest.fixture(scope='session',
+                params=['binary', 'multiclass'])
+def regression_dataset(request):
+    regression_type = request.param
+    n_samples = 500000
+    n_features = 5
+
+    data = (np.random.rand(n_samples, n_features) * 2) - 1
+
+    if regression_type == 'binary':
+        coef = (np.random.rand(n_features) * 2) - 1
+        coef /= np.linalg.norm(coef)
+        output = (data @ coef) > 0
+    elif regression_type == 'multiclass':
+        n_classes = 3
+        coef = (np.random.rand(n_features, n_classes) * 2) - 1
+        coef /= np.linalg.norm(coef, axis=0)
+        output = (data @ coef).argmax(axis=1)
+
+    output = output.astype(np.int32)
+    return regression_type, data, coef, output
+
+
+@pytest.mark.parametrize('option', ['sample_weight', 'class_weight',
+                                    'balanced', 'no_weight'])
+def test_logistic_regression_weighting(regression_dataset, option):
+    regression_type, data, coef, output = regression_dataset
+
+    if option == 'sample_weight':
+        n_samples = data.shape[0]
+        sample_weight = np.abs(np.random.rand(n_samples))
+
+        culog = cuLog(fit_intercept=False)
+        culog.fit(data, output, sample_weight=sample_weight)
+
+        sklog = skLog(fit_intercept=False)
+        sklog.fit(data, output, sample_weight=sample_weight)
+    elif option == 'class_weight':
+        class_weight = np.random.rand(2)
+        class_weight = {0: class_weight[0], 1: class_weight[1]}
+
+        culog = cuLog(fit_intercept=False, class_weight=class_weight)
+        culog.fit(data, output)
+
+        sklog = skLog(fit_intercept=False, class_weight=class_weight)
+        sklog.fit(data, output)
+    elif option == 'balanced':
+        culog = cuLog(fit_intercept=False, class_weight='balanced')
+        culog.fit(data, output)
+
+        sklog = skLog(fit_intercept=False, class_weight='balanced')
+        sklog.fit(data, output)
+    else:
+        culog = cuLog(fit_intercept=False)
+        culog.fit(data, output)
+
+        sklog = skLog(fit_intercept=False)
+        sklog.fit(data, output)
+
+    skcoef = np.squeeze(sklog.coef_)
+    cucoef = np.squeeze(culog.coef_)
+    if regression_type == 'binary':
+        skcoef /= np.linalg.norm(skcoef)
+        cucoef /= np.linalg.norm(cucoef)
+        unit_tol = 0.04
+        total_tol = 0.08
+    elif regression_type == 'multiclass':
+        skcoef = skcoef.T
+        skcoef /= np.linalg.norm(skcoef, axis=1)[:, None]
+        cucoef /= np.linalg.norm(cucoef, axis=1)[:, None]
+        unit_tol = 0.2
+        total_tol = 0.3
```

Comment:
It would be good to add a comment of how the tolerances have been derived, same for binary

Reply:
Well, to be frank there are derived from manual iterative testing...